### ISSUE TITLE
Word Wrap on Identify Feature

### ISSUE THREAD

**Issue by lbell** [qgis/QGIS#31650]
**Feature description.**
If you're identifying a feature, long text fields are represented as a single-line. Would be nice to be able to word-warp those lines to see everything clearly.

**Comment by rscircus** [comment]
Hi @lbell - I think, I can fix this. Can you provide me with a screenshot?

**Comment by lbell** [comment]
Thanks! This occurs in both the "Form" view as well as the "Identify Results"  panel. Both depicted below.

![image](https://user-images.githubusercontent.com/1029014/65432299-326a1c00-ddd8-11e9-8b38-f1db1cac5532.png)

**Comment by rscircus** [comment]
https://github.com/rscircus/QGIS/tree/feature/WordWrapIdentifyResults

**Comment by rscircus** [comment]
OK, it took me a while to set this up, following this: https://htmlpreview.github.io/?https://raw.github.com/qgis/QGIS/master/doc/INSTALL.html - however, it's compiling now on 4 cores and I get a coffee. With a little bit of luck, this will be done today evening at least for the 'Identify Results' part.

**Comment by rscircus** [comment]
@lbell - could you send me your QGIS version?

**Comment by lbell** [comment]
QGIS version | 3.8.3-Zanzibar | QGIS code revision | 685d8b15d2
-- | -- | -- | --
Compiled against Qt | 5.9.5 | Running against Qt | 5.9.5
Compiled against GDAL/OGR | 2.4.2 | Running against GDAL/OGR | 2.4.2
Compiled against GEOS | 3.7.1-CAPI-1.11.1 | Running against GEOS | 3.7.1-CAPI-1.11.1 27a5e771
PostgreSQL Client Version | 10.10 (Ubuntu 10.10-0ubuntu0.18.04.1) | SpatiaLite Version | 4.3.0a
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.2
Compiled against PROJ | 5.2.0 | Running against PROJ | Rel. 5.2.0, September 15th, 2018
OS Version | Ubuntu 18.04.3 LTS |   |

**Comment by rscircus** [comment]
Would it be OK for you, if I pack the full text into the tooltip?

**Comment by elpaso** [comment]
Try checking the widget "multiline" option in the form configuration for that field, it should word-wrap automatically.

**Comment by rscircus** [comment]
@elpaso , damn it. I thought that's a nice way to get into the code base. And now I hear it's solved already. :see_no_evil:

**Comment by rscircus** [comment]
![image](https://user-images.githubusercontent.com/1167114/65537397-6e4ecf80-df05-11e9-9a8d-928c4b6465ce.png)

There it is. In Layer -> Properties -> Attributes Form.

**Comment by rscircus** [comment]
@elpaso - you close this?

**Comment by elpaso** [comment]
@rscircus hey, don't loose the momentum, there is a long list of issues if you are in bugfixing mode :smile_cat:

**Comment by DelazJ** [comment]
> Try checking the widget "multiline" option in the form configuration for that field, it should word-wrap automatically.

Imho, this kind of information is worth a mention in [docs](https://docs.qgis.org/testing/en/docs/user_manual/working_with_vector/vector_properties.html#edit-widgets)

**Comment by lbell** [comment]
Almost Great! 
1) That only works on the widget -- though it is a bit tedious to turn that on for every field that exceeds a certain length... on every layer... potentially hundreds of clicks to do something that could be accomplished with "if this field exceeds XX characters, wrap it automatically"

2) I still do not see a way to word-wrap the 'Identify features' panel (right hand side of my screenshot above). This is the most vital as I'm clicking through a whole bunch of features on a large project -- many of which I man not know how long the fields are, and I usually don't want the widget popping up in my way when I am clicking through things quickly.

@elpaso Can we keep this enhancement ticket open till this is addressed?

**Comment by elpaso** [comment]
Sure, in theory, the identify widget should honor the options configured in the field config (through field formatters IIRC), we need to check why it does not in that case.

**Comment by elpaso** [comment]
Ok, the fix is trivial, but I'm not 100% sure we want this:

![image](https://user-images.githubusercontent.com/142164/65617849-afa2b600-dfbd-11e9-9d27-b667eeb31af4.png)

```diff
--- a/src/app/qgsidentifyresultsdialog.cpp
+++ b/src/app/qgsidentifyresultsdialog.cpp
@@ -618,14 +618,17 @@ void QgsIdentifyResultsDialog::addFeature( QgsVectorLayer *vlayer, const QgsFeat
     if ( foundLinks )
     {
       QLabel *valueLabel = new QLabel( links );
+      valueLabel->setWordWrap( true );
       valueLabel->setOpenExternalLinks( true );
       attrItem->setData( 1, Qt::DisplayRole, QString() );
       attrItem->treeWidget()->setItemWidget( attrItem, 1, valueLabel );
     }
     else
     {
-      attrItem->setData( 1, Qt::DisplayRole, value );
-      attrItem->treeWidget()->setItemWidget( attrItem, 1, nullptr );
+      auto valueLabel { qgis::make_unique<QLabel>( value ) };
+      valueLabel->setWordWrap( true );
+      attrItem->setData( 1, Qt::DisplayRole, QString() );
+      attrItem->treeWidget()->setItemWidget( attrItem, 1, valueLabel.release() );
     }
 
     if ( fields.at( i ).name() == vlayer->displayField() )
```

**Comment by elpaso** [comment]
@nirvn @nyalldawson you are the UX master, what do you think?

**Comment by lbell** [comment]
Seems arbitrary to close this.

**Comment by DelazJ** [comment]
I'd tend to agree. The issuer is not the one to provide feedback. So let's reopen.

**Comment by idantene** [comment]
FWIW, it's been several years, and it's still not implemented.

@elpaso I think in terms of UX, this should simply be a toggle-able option in the "Identify Results" pane, similarly to how you can choose to "Hide derived attributes from results".

**Comment by AddisonSnyder** [comment]
Agreed, would be nice to have a toggle. But also the workaround doesn't actually word-wrap so much as interpret newlines. While a lot better, I still need to expand the pane an obscene amount to actually read long lines.

**Comment by ajsnyde** [comment]
Agreed, would be nice to have a toggle. But also the workaround doesn't actually word-wrap so much as interpret newlines. While a lot better, I still need to expand the pane an obscene amount to actually read long lines.

**Comment by merydian** [comment]
How about adding a scroll area to the label, then making a checkbox in the settings for that (In the wrench button in the `Identify Results` window)? 

<img width="527" height="614" alt="Image" src="https://github.com/user-attachments/assets/5572dd45-c993-4266-a077-a074f656b76a" />

I've managed to set `valueLabel->setWordWrap(true);`, but there's a problem with rendering the full text with that on its own.

```cpp
if ( foundLinks )
{
  QLabel *valueLabel = new QLabel( links );
  valueLabel->setOpenExternalLinks( true );
  valueLabel->setWordWrap(true);

  QScrollArea *scrollArea = new QScrollArea();
  scrollArea->setWidget(valueLabel);
  scrollArea->setWidgetResizable(true);
  scrollArea->setFrameStyle( QFrame::NoFrame );

  attrItem->setData( 1, Qt::DisplayRole, QString() );
  QTreeWidget *tw = attrItem->treeWidget();
  tw->setItemWidget( attrItem, 1, scrollArea );
}
else
{
  QLabel *valueLabel = new QLabel( representedValue );
  valueLabel->setWordWrap(true);

  QScrollArea *scrollArea = new QScrollArea();
  scrollArea->setWidget(valueLabel);
  scrollArea->setWidgetResizable(true);
  scrollArea->setFrameStyle( QFrame::NoFrame );

  attrItem->setData( 1, Qt::DisplayRole, QString() );
  QTreeWidget *tw = attrItem->treeWidget();
  tw->setItemWidget( attrItem, 1, scrollArea );
}
```
